pkg/storage-init/storage-init.sh

```diff
@@ -57,43 +57,27 @@ zfs_set_arc_limits() {
 }
 
 # should be in sync with options inside installer
-zfs_module_load() {
-    zfs_options="zfs_compressed_arc_enabled=0 \
-zfs_vdev_min_auto_ashift=12 \
-zvol_request_sync=0 \
-zfs_vdev_aggregation_limit_non_rotating=$(( 1024*1024 )) \
-zfs_vdev_async_write_active_min_dirty_percent=10 \
-zfs_vdev_async_write_active_max_dirty_percent=30 \
-zfs_delay_min_dirty_percent=40 \
-zfs_delay_scale=800000 \
-zfs_dirty_data_sync_percent=15 \
-zfs_prefetch_disable=1 \
-\
-zfs_vdev_sync_read_min_active=35 \
-zfs_vdev_sync_read_max_active=35 \
-zfs_vdev_sync_write_min_active=35 \
-zfs_vdev_sync_write_max_active=35 \
-zfs_vdev_async_read_min_active=1 \
-zfs_vdev_async_read_max_active=10 \
-zfs_vdev_async_write_min_active=1 \
-zfs_vdev_async_write_max_active=10 \
-\
-zfs_smoothing_scale=50000 \
-zfs_smoothing_write=5
-"
-
-    # Disabling SC2086 because word splitting here is intended -
-    # otherwise modprobe would not recognize arguments
-    # shellcheck disable=SC2086
-    if ! modprobe zfs ${zfs_options}; then
-        echo "Failed to load ZFS module with parameters, falling back to defaults"
-        modprobe zfs
-    fi
-}
-
-zfs_module_unload() {
-    # shellcheck disable=SC2046
-    rmmod $(lsmod | grep zfs | awk '{print $1;}') || :
+zfs_set_default_parameters() {
```

Comment:
My understanding was that zfs needed some of these set at load time for them to take effect. Perhaps @zededa-yuri or @zedi-pramodh can confirm and indicate how one checks which settings are used by the running ZFS code.

Reply:
Since zfs is now a built-in nothing to load/unload. /sys/module/zfs entries are always there.

Reply:
Yes, but I thought zfs allocates datastructures based on some of the parameters when it is initialized, which currently happens when it is loaded. Thus we need to make sure that all of the settings done after it is initialized indeed take effect. Maybe I'm confused and we just needed to make sure they are set before we create the first pool and vdev (in the installer and/or storage-init). Can you ping @zededa-yuri and @zedi-pramodh to check?

Reply:
The usual practice is to do nothing on module load, but to prepare memory caches for common structures, register sysfs/procfs entries and do all these sort of preparations. Real settings will take effect once something is mounted. Quick look into the openzfs module init functions does not show me that zfs behaves differently - settings are not taken into consideration on module load. But of course I can ask Pramodh / Yuri to verify my assumption.

Reply:
@rouming can you help me understand zfs specific changes here. Are you removing those zfs  parameters set during zfs load time ? We need those parameters set when zfs is starting because those are tested and tuned for EVE workload and hence always set when system is initiated or rebooted. Those are not default values.

Reply:
@zedi-pramodh nope, all parameters are set as before. The only difference is that now there is no zfs module, zfs is built-in, so no need to modprobe/rmmod and parameters are set through the sysfs module entry. Once kernel is booted we set these parameters and then call mount or create zfs pool, etc.

Reply:
If parameters are not defined as options to load module, probably we can avoid duplicate here and in insrall script and move such options to somewhere in onboot logic? Maybe as a part of https://github.com/lf-edge/eve/tree/master/pkg/dom0-ztools/rootfs/etc/init.d.

Reply:
That's a good point. What if instead of another shell custom script invention we can pass zfs parameters through the boot config? https://docs.kernel.org/admin-guide/bootconfig.html or that's too much?

Reply:
@rouming I am fine whatever approach we take as long as those parameters are set whenever system starts. But remember not all systems are deployed with zfs most of them are ext4. So those parameters should be set on zfs systems only. Also, when you say zfs is built-in, are we making sure all the patches we have for zfs /eve/pkg/kernel/patches-zfs-2.1.2/ are compile into it ?

Reply:
Yes, all patches are there, because the preparation process of zfs sources was not touched.

Reply:
Oh basically you just removed modprobe/rmmod because zfs is built in, and rest of the zfs setup remains same, so then I do not see any issue.